app/assets/javascripts/index/directions-endpoint.js

```diff
@@ -79,12 +111,32 @@ OSM.DirectionsEndpoint = function Endpoint(map, input, iconUrl, dragCallback, ch
 
       setLatLng(L.latLng(json[0]));
 
+      endpoint.value = json[0].display_name;
       input.val(json[0].display_name);
 
       changeCallback();
     });
   }
 
+  function getReverseGeocode() {
+    var latlng = endpoint.latlng.clone();
+    var reverseGeocodeUrl = OSM.NOMINATIM_URL + "reverse?lat=" + latlng.lat + "&lon=" + latlng.lng + "&format=json";
```

Comment:
Do we need to control the match precision here? Might this cause a location to "snap" to something a long way away?

Reply:
The location wouldn't snap in the same way is does on the currently deployed version. The pointer is not going to move, just the address would be too imprecise.

Ultimately we should control the precision but I didn't want to do it in this pull request, see https://github.com/openstreetmap/openstreetmap-website/pull/4904#discussion_r1723165682